### PR TITLE
Added derivative of scalar factor in utils

### PR DESCRIPTION
**Pull request by ritzvik**

### CODE REVIEW

**Comment by shreyasbapat** [comment]
Quality issues @ritzvik

File: src/einsteinpy/metric/schwarzschild.py [hunk 1]
```diff
@@ -36,9 +36,9 @@ def from_spherical(cls, pos_vec, vel_vec, time, M):
 
         Parameters
         ----------
-        pos_vector : list
+        pos_vec : list
```
Comment (shreyasbapat) [inline src/einsteinpy/metric/schwarzschild.py]: Great

File: src/einsteinpy/utils.py [hunk 2]
```diff
@@ -26,18 +26,19 @@ def schwarzschild_radius(mass):
...
     Flat Universe.
 
     Parameters
     ----------
     era : string
-        Can be chosen from md (Matter Dominant),
-        rd (Radiation Dominant) and ded (Dark Energy Dominant)
-    t : float
-        Time for the event (Value in range [0,1])
+        Can be chosen from 'md' (Matter Dominant),
```
Comment (shreyasbapat) [inline src/einsteinpy/utils.py]: So correct

File: src/einsteinpy/utils.py [hunk 3]
```diff
@@ -51,13 +52,53 @@ def scalar_factor(era, t, tuning_param=1):
...
     elif era == "rd":
-        return tuning_param * (t ** (0.5))
+        return tuning_param * (T ** (0.5))
     elif era == "ded":
         hubble_const = (constant.Cosmo_Const / 3) ** 0.5
-        val = np.e ** (hubble_const * t)
+        val = np.e ** (hubble_const.value * T)
+        return tuning_param * val
+    else:
+        raise ValueError("Passed era is out of syllabus!")
```
Comment (shreyasbapat) [inline src/einsteinpy/utils.py]: @ritzvik  How about formalizing this one? Maybe this will be too much for professional users.

**Comment by spino17** [comment]
How about you merge it and commit the changes yourselves. You yourself made the class initially 😂.

**Comment by spino17** [comment]
@ritzvik commenting from @spino17 's profile

**Comment by ritzvik** [comment]
code professional enough?

File: src/einsteinpy/utils.py [hunk 4]
```diff
@@ -51,16 +52,56 @@ def scalar_factor(era, t, tuning_param=1):
...
-        return tuning_param * (t ** (0.5))
+        return tuning_param * (T ** (0.5))
     elif era == "ded":
         hubble_const = (constant.Cosmo_Const / 3) ** 0.5
-        val = np.e ** (hubble_const * t)
+        val = np.e ** (hubble_const.value * T)
         return tuning_param * val
     else:
-        raise ValueError("Passed era is out of syllabus!")
+        raise ValueError("Passed era should be either 'md', 'rd' or 'ded' ")
```
Comment (shreyasbapat) [inline src/einsteinpy/utils.py]: Amazing

**Comment by shreyasbapat** [comment]
Merging! 
Meanwhile Please remember to mention every issue a pull request addresses.

#15

**Pull request merged**